backend/typescript/rest/schedulingRoutes.ts

```diff
@@ -34,26 +34,37 @@ const schedulingService: ISchedulingService = new SchedulingService(
 
 /* Get all schedulings, optionally filter by:
   - id, through URI (ex. /scheduling/1)
-  - donorId, through query param (ex. /scheduling/?donorId=1)
+  - donorId or volunteerId, through query param (ex. /scheduling/?donorId=1)
 */
 schedulingRouter.get("/:id?", async (req, res) => {
   const { id } = req.params;
-  const { donorId } = req.query;
-  const { weekLimit } = req.query;
+  const { donorId, volunteerId, weekLimit, volunteerNeeded } = req.query;
   const contentType = req.headers["content-type"];
 
-  if (id && donorId) {
+  if (
+    (id && donorId) ||
+    (id && volunteerId) ||
+    (donorId && volunteerId) ||
+    (id && volunteerNeeded) ||
+    (volunteerId && volunteerNeeded) ||
+    (donorId && volunteerNeeded)
+  ) {
     await sendResponseByMimeType(res, 400, contentType, [
       {
-        error: "Cannot query by both id and donorId.",
+        error: "Cannot query by multiple parameters.",
       },
     ]);
     return;
   }
 
-  if (!id && !donorId) {
+  const getScheduleType =
+    volunteerNeeded !== undefined ? volunteerNeeded === "true" : undefined;
```

Comment:
as mentioned on discord call, could remove this section and just check [since the only case we'll ever send in `volunteerNeeded` it'll be true and it'll be for the page when potential volunteers want to see availble shifts

```typescript
if (volunteerNeeded) {
try {
const schedulings = await schedulingService.getSchedulesThatNeedVolunteers();
...

}
}
```

Reply:
so then we keep `getSchedulings()` to just what we have right now (gets all schedules with no queries)